homeassistant/components/sensor/swiss_public_transport.py

```diff
@@ -0,0 +1,148 @@
+    """ Get the Swiss public transport sensor. """
+
+    if None in (hass.config.latitude, hass.config.longitude):
+        _LOGGER.error("Latitude or longitude not set in Home Assistant config")
+        return False
+
+    try:
+        # pylint: disable=unused-variable
+        from requests import get
+
+    except ImportError:
+        _LOGGER.exception(
+            "Unable to import requests. "
+            "Did you maybe not install the 'Requests' package?")
+
+        return None
+
+    # journal contains [0] Station ID start, [1] Station ID destination
+    # [2] Station name start, and [3] Station name destination
+    journey = []
+    journey.append(config.get('from', None))
+    journey.append(config.get('to', None))
+    try:
+        for location in [config.get('from', None), config.get('to', None)]:
+            # transport.opendata.ch doesn't play nice with requests.Session
+            result = get(_RESOURCE + 'locations?query=%s' % location)
+            journey.append(result.json()['stations'][0]['name'])
+    except KeyError:
+        _LOGGER.error(
+            "Unable to determine stations. "
+            "Check your settings and/or the availability of opendata.ch")
+
+        return None
+
+    dev = []
+    data = PublicTransportData(journey)
+    dev.append(SwissPublicTransportSensor(data, journey))
+    add_devices(dev)
+
+
+# pylint: disable=too-few-public-methods
+class SwissPublicTransportSensor(Entity):
+    """ Implements an Swiss public transport sensor. """
+
+    def __init__(self, data, journey):
+        self.data = data
+        self._name = journey[2] + '-' + journey[3]
+        self.update()
+
+    @property
+    def name(self):
+        """ Returns the name. """
+        return self._name
+
+    @property
+    def state(self):
+        """ Returns the state of the device. """
+        return self._state
+
+    # pylint: disable=too-many-branches
+    def update(self):
+        """ Gets the latest data from opendata.ch and updates the states. """
+        times = self.data.update()
+        if times is not None:
+            self._state = times[0] + ', ' + times[1]
+
+
+# pylint: disable=too-few-public-methods
+class PublicTransportData(object):
+    """ Class for handling the data retrieval.  """
+
+    def __init__(self, journey):
+        self.times = ['n/a', 'n/a']
+        self.start = journey[0]
+        self.destination = journey[1]
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """ Gets the latest data from opendata.ch. """
+
+        from requests import get
+
+        response = get(
+            _RESOURCE +
+            'connections?' +
+            'from=' + self.start + '&' +
+            'to=' + self.destination + '&' +
+            'fields[]=connections/from/departureTimestamp/&' +
+            'fields[]=connections/')
+
+        try:
+            self.times.insert(0, dt_util.timestamp_to_short_time_str(
```

Comment:
Is there a reason you're caching all the old results ? Right now this will keep adding to the list.

Based on how you use the data, it looks like you want something like this:
```python
connections = response.json()['connections'][:2]

try:
  return [
    dt_util.timestamp_to_short_time_str(item['from'])
    for item in connections
  ]
except KeyError:
  return ['n/a']
```